pymeasure/instruments/common_base.py

```diff
@@ -193,6 +188,24 @@ class attributes that are a DynamicProperty. Check also for class variables
                     setattr(self, self.__reserved_prefix + attr, obj.__dict__[attr])
         return special_names
 
+    def _create_channels(self):
+        """Create channels according to the ChannelCreator objects.
+
+        If a subclass overrides a ChannelCreator object with another one, the
+        one of the subclass is used, not that of the superclass.
+        """
+        channel_creators = {}
+        # Collect all ChannelCreators, override in order of subclasses
+        for obj in reversed((self,) + self.__class__.__mro__):
```

Comment:
I'm nervous that we need to use `__mro__`, to be honest :sweat_smile:
The override should automatically be solved by a lookup -- since the MRO starts with the bottom-most class, if it finds a wanted attribute, it just stops. Why do we need to emulate this, here?

Second question, why are you prepending `self` to the mro list. On the one had, this seems to mix an instantiated object with the `<class>` objects that should be in the `__mro__` list. On the other hand, the class itself should already be the first entry in the MRO??

Reply:
Blind code sketch what I mean:
```
            for item, value in self.__dict__.items():
                if isinstance(value, self.ChannelCreator):
                    channel_creators[item] = self.gettattr(value)  # let the MRO find the correct attribute (most derived class first)
```

Reply:
If you give me a possibility to use automatic lookup, I'll use it.
The instance (`self`) dir does not contain them, and the class dirs do not inherit either.
Therefore, you have to look in the instance `self` (in case someone added it to the instance) , the class and every superclass (answering question 2).

I could use `dir()` and `getattr`, which would raise errors with `setting `.

Reply:
Neither `Instrument.setting` nor the special values of dynamic properties like to be read, but we could catch these errors with try clauses.

Reply:
If you find such a possibility, we Don't need the dictionary anymore, but you can directly create the channels. I use the dictionary to search for the most recent definition before I add them to the instrument.

Reply:
What about using inspect? It looks cleaner to me.
I think we have 3 places in the code base where such lookup is performed:
- parameters (using inspect)
- dynamic properties (using MRO)
- channel creator

I think once the best approach is selected, we could make the approach uniform.

Reply:
`inspect.getmembers` fails, if you either have a dynamic property or a setting property, because the first one creates protected names (which cannot be read) and the latter cannot return a value either.

I based the code on that of dynamic properties, but I reversed the resolution order in order, that the most recent definition remains (as expected).

We could do it in the right order (from subclass to superclass), if we used `setdefault`, but I deemed direct assign better.

Reply:
I think, that `inspect` doesn't fails, because we are reading attribute at class level, something like:
`for attr, value in inspect.getmembers(self.__class__):`
In this case, python properties are not evaluated and value returned is a property object.

Reply:
Ah, getting the members of the class is great, thanks. I'll try it at home.

Shall I change special names as well to use inspect?

Reply:
Make sense to me.

Reply:
> Shall I change special names as well to use inspect?

Yes, please. Also take into account that special names uses also `self` in the attribute lookup, which means that take into account also instance, but I believe this is a bug and so search in class only is also fixing this.

Reply:
Change is implemented (together with additional tests, to verify, that dynamic properties continue to work as expected).